holidays/countries/cayman_islands.py

```diff
@@ -0,0 +1,166 @@
+
+        # Discovery Day.
+        self._add_holiday_3rd_mon_of_may(tr("Discovery Day"))
+
+        if self._year <= 2022:
+            queens_birthday_dates = {
+                2008: (JUN, 16),
+                2009: (JUN, 15),
+                2010: (JUN, 14),
+                2011: (JUN, 13),
+                2012: (JUN, 18),
+                2013: (JUN, 17),
+                2014: (JUN, 16),
+                2015: (JUN, 15),
+                2016: (JUN, 13),
+                2017: (JUN, 19),
+                2018: (JUN, 11),
+                2019: (JUN, 10),
+                2020: (JUN, 15),
+                2021: (JUN, 14),
+                2022: (JUN, 6),
+            }
+            # Queen's Birthday.
+            name = tr("Queen's Birthday")
+            self._add_holiday(name, queens_birthday_dates[self._year])
+
+        if self._year >= 2023 and self._year <= 2025:
+            kings_birthday_dates = {
+                2023: (JUN, 19),
+                2024: (JUN, 17),
+                2025: (JUN, 23),
+            }
+            # King's Birthday.
+            self._add_holiday(tr("King's Birthday"), kings_birthday_dates[self._year])
+
+        # Constitution Day.
+        self._add_holiday_1st_mon_of_jul(tr("Constitution Day"))
+
+        # Remembrance Day.
+        self._add_holiday_2nd_mon_of_nov(tr("Remembrance Day"))
+
+        self._add_observed(
+            # Christmas Day.
+            self._add_christmas_day(tr("Christmas Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE,
+        )
+
+        self._add_observed(
+            # Boxing Day.
+            self._add_christmas_day_two(tr("Boxing Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE,
+        )
+
+
+class KY(CaymanIslands):
+    pass
+
+
+class CYM(CaymanIslands):
+    pass
+
+
+class CaymanIslandsStaticHolidays:
+    """Cayman Islands special holidays.
+
+    References:
+        * [2009 Cayman Islands Constitution Day](https://web.archive.org/web/20250320152628/https://www.constitutionalcommission.ky/the-cayman-islands-constitution-2009)
+        * [UK Royal Wedding](https://en.wikipedia.org/wiki/Wedding_of_Prince_William_and_Catherine_Middleton)
+        * [Queen Elizabeth II's Diamond Jubilee](https://web.archive.org/web/20250708170324/https://calendarific.com/holiday/cayman-islands/queen-diamond-jubilee)
+        * [Queen Elizabeth II's Funeral](https://www.gov.ky/queen-elizabeth-ii/faqs)
+        * [King Charles III's Coronation](https://web.archive.org/web/20250601214328/https://www.radiocayman.gov.ky/news/public-holidays-for-2023-unconfirmed)
+    """
+
+    special_public_holidays = {
+        # 2009 Cayman Islands Constitution Day.
+        2009: (NOV, 6, tr("2009 Cayman Islands Constitution Day")),
+        # UK Royal Wedding.
+        2011: (APR, 29, tr("UK Royal Wedding")),
+        # Queen Elizabeth II's Diamond Jubilee.
+        2012: (JUN, 4, tr("Queen Elizabeth II's Diamond Jubilee")),
```

Comment:
According to [the 2012-2013 official calendar](https://archive.ph/2012.12.20-043003/http://www.gov.ky/portal/page#selection-291.3-291.32), there's a "Referendum Day" on July 18th, 2012 as well; [Cross-Ref Here](https://web.archive.org/web/20120711005705/http://www.cayman27.com.ky/2012/06/21/chamber-against-referendum-holiday)

Also a missing "General Election Day" entry for May 22nd, 2013